docs/user_guide/assessment.rst

```diff
@@ -589,6 +589,94 @@ For more examples, please
 see the :ref:`sphx_glr_auto_examples_plot_new_metrics.py` notebook in the
 :ref:`examples`.
 
+.. _bootstrap:
+
+Confidence Intervals with Bootstrapping
+---------------------------------------
+
+It is a truth universally acknowledged, that a person in possession of
+a statistic, must be in want of an error bar.
+The only question is whether or not the person is aware of their want.
+Two aspects of fairness metrics make this particularly acute:
+
+1. Some of the subgroups defined by the sensitive features may be much
+   smaller than the overall sample size, leading to high levels of
+   random noise
+2. We often then wish to measure the disparity between two such subgroups.
+   While this means taking a difference or a ratio of the metrics,
+   the errors will always add (in quadrature)
+
+To address the need for error estimates, :class:`MetricFrame` provides
+a means of estimating confidence intervals for scalar metrics by using
+a process known as bootstrapping.
+This avoids the need for an analytic form for the error to be provided
+for each metric computed.
+
+Bootstrapping works as follows:
+
+1. Generate a large number of synthetic datasets by sampling the
+   target dataset *with replacement* (i.e. a given sample from the target
+   dataset may appear more than once in a synthetic dataset)
+2. Compute the metric value for each of these synthetic datasets
+3. Fit a distribution to this set of metric values
+4. Obtain the desired confidence intervals from that distribution
+
+Let us look at an example:
+
+.. doctest:: assessment_metrics
+    :options:  +NORMALIZE_WHITESPACE
+
+    >>> # Enable bootstrapping with the n_boot parameter
+    >>> metric_ci = MetricFrame(metrics=skm.accuracy_score,
+    ...                         y_true=decision,
+    ...                         y_pred=prediction,
+    ...                         sensitive_features={'SF' : sensitive_feature},
+    ...                         n_boot=100)
+    >>> metric_ci.overall
+    0.4
+    >>> metric_ci.overall_ci
+    [(0.025, 0.23333333333333334), (0.975, 0.6)]
+
+We have turned on bootstrapping by setting the :code:`n_boot` parameter;
+in this case, we have requested that 100 synthetic datasets be generated
+to create the statistics.
+We see that the :attr:`MetricFrame.overall` property is unchanged.
+However, we can also access the :attr:`MetricFrame.overall_ci` property
+which contains estimates of the confidence interval.
+By default, the bootstrapping algorithm computes the 95% confidence
+interval, which ranges from the 25th millile to the 975th millile.
```

Comment:
Feedback on whether 'millile' is the right word invited.

Reply:
Or should I just go with 'quantile' and give decimals?

Reply:
Will do. Although this particular part of the PR is not going to survive the other changes I'm making to the MetricFrame documentation.